src/analytics/ValoraAnalytics.test.ts

```diff
@@ -152,6 +143,21 @@ const defaultProperties = {
 
 describe('ValoraAnalytics', () => {
   let ValoraAnalytics: typeof ValoraAnalyticsModule
+  const mockPromiseFactory = () =>
+    jest.fn().mockReturnValue({
+      then: jest.fn(),
+      catch: jest.fn(),
+    })
+  const mockSegmentClient = {
+    identify: mockPromiseFactory(),
+    track: mockPromiseFactory(),
+    screen: mockPromiseFactory(),
+    flush: mockPromiseFactory(),
```

Comment:
nit: You can just do `jest.fn().mockResolvedValue(undefined)`  instead of the promise factory right?